www/src/components/account/billing/BillingUpgradeToProfessionalModal.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import { Link } from 'react-router-dom'
+import { useMutation } from '@apollo/client'
+import { Div, Flex } from 'honorable'
+import { Button, Modal } from '@pluralsh/design-system'
+
+import PlatformPlansContext from '../../../contexts/PlatformPlansContext'
+import BillingBankCardContext from '../../../contexts/BillingBankCardContext'
+import BillingConsumptionContext from '../../../contexts/BillingConsumptionContext'
+import SubscriptionContext from '../../../contexts/SubscriptionContext'
+
+import { UPGRADE_TO_PROFESSIONAL_PLAN_MUTATION } from './queries'
+
+import useBankCard from './useBankCard'
+
+import BillingPreview from './BillingPreview'
+import BillingError from './BillingError'
+
+type BillingUpgradeToProfessionalModalPropsType = {
+  open: boolean
+  onClose: () => void
+}
+
+function BillingUpgradeToProfessionalModal({ open, onClose }: BillingUpgradeToProfessionalModalPropsType) {
+  const { proPlatformPlan } = useContext(PlatformPlansContext)
+  const { nClusters, nUsers } = useContext(BillingConsumptionContext)
+  const { card } = useContext(BillingBankCardContext)
```

Comment:
You should be able to determine this w/o fetching the card. The definitive mechanism is if the account has a non null `customerId` attribute

Reply:
I'm fetching the card simply to display the card on the payment modal, not for the mutation.